product_supplierinfo_for_customer_sale/models/sale.py

```diff
@@ -48,12 +41,46 @@ def product_id_change(self):
                 ('applied_on', '=', '0_product_variant'),
             ])
             if items:
-                code_id = self.env['product.supplierinfo'].search([
-                    '|', ('product_tmpl_id', '=', product.product_tmpl_id.id),
-                    ('product_id', '=', product.id),
-                    ('supplierinfo_type', '=', 'customer'),
-                    ('name', '=', line.order_id.partner_id.id),
-                ], limit=1)
-                if code_id and code_id.min_qty:
-                    line.product_uom_qty = code_id.min_qty
+                supplierinfo = self.get_supplierinfo(line)
+                line.product_uom_qty = supplierinfo and supplierinfo.min_qty
         return result
+
+    def get_supplierinfo(self, line):
+        """
+        Search supplierinfo only by product template if it has not variant for
+        performance.
+        """
+        if line.product_id.product_variant_count > 1:
+            return self._search_supplierinfo_by_variant(line)
```

Comment:
You should always search by this condition, as there could be in the past several variants that are now deactivated by any reason and their supplierinfo records stay there yet, and you are returning the incorrect one.

Reply:
@sergio-teruel what about this comment?

Reply:
This comment is still pending

Reply:
Ok, in which one are you prefer?